tests/integration/internal/test_lock.py

```diff
@@ -0,0 +1,103 @@
+import random
+import threading
+
+from spccore.internal.lock import *
+
+
+def test_private_acquire():
+    user1_lock = Lock("foo", max_age=datetime.timedelta(seconds=5))
+    user2_lock = Lock("foo", max_age=datetime.timedelta(seconds=5))
+
+    assert user1_lock._acquire_lock()
+    assert user1_lock._get_age() < 5
+    assert user2_lock._acquire_lock() is False
+
+    user1_lock.release()
+
+    assert user2_lock._acquire_lock()
+    assert user1_lock._acquire_lock() is False
+
+    user2_lock.release()
+
+
+def test_context_manager():
+    user1_lock = Lock("foo", max_age=datetime.timedelta(seconds=5))
+    user2_lock = Lock("foo", max_age=datetime.timedelta(seconds=5))
+
+    with user1_lock:
+        assert user1_lock._get_age() < 5
+        assert user2_lock._acquire_lock() is False
+
+    with user2_lock:
+        assert user2_lock._acquire_lock()
+        assert user1_lock._acquire_lock() is False
+
+
+def test_lock_timeout():
+    user1_lock = Lock("foo", max_age=datetime.timedelta(seconds=1))
+    user2_lock = Lock("foo", max_age=datetime.timedelta(seconds=1))
+
+    with user1_lock:
+        assert user1_lock.held
+        assert user1_lock._get_age() < 1.0
+        assert user2_lock._acquire_lock(break_old_locks=True) is False
+        time.sleep(1.1)
+        assert user1_lock._get_age() > 1.0
+        assert user2_lock._acquire_lock(break_old_locks=True)
+
+
+def test_renew():
+    user1_lock = Lock("foo", max_age=datetime.timedelta(seconds=1))
+    user2_lock = Lock("foo", max_age=datetime.timedelta(seconds=1))
+
+    with user1_lock:
+        time.sleep(1.1)
+        assert user1_lock._get_age() > 1.0
+        user1_lock.renew()
+        assert user1_lock._get_age() < 1.0
+        assert user2_lock._acquire_lock(break_old_locks=True) is False
+        time.sleep(1.1)
+        assert user1_lock._get_age() > 1.0
+        assert user2_lock._acquire_lock(break_old_locks=True)
+
```

Comment:
may not be able to use 'with' when the enclosed code block is not << the lock duration.